sandbox/errorHandling.lua

```diff
@@ -1,48 +1,35 @@
--- package name possibly should be debug
--- but this clashes with a stock lua package
+-- the package name "debug" debatably may be better than errorHandling
+-- but there is a stock lua package with that name (which isnt included in Empty Epsilon)
 errorHandling = {}
 
+-- please note - this file is likely to become either irrelevant or in need of update when the EE ECS rework is complete
+-- it would be nice if the whole walking through the class hierarchy was moved out into its own file
+-- however that walking is likely to be made obsolete semi soon with the ECS system
+
+-- note some SpaceObject are never wrapped - a good example would be missiles fired from craft
+-- in general there is a fair chance anything created inside of the EE engine will not have these functions wrapped
+-- there are a few soultions
+-- we could run each update and find unwrapped objects
+-- we could make a pull request to master for an onNewSpaceObject
+-- the latter is better in my eyes
+-- I am planning personally on waiting ECS changes have been implemented and I'm more sure on whats happening
+-- PlayerShips created via the gm screen are ***NOT*** wrapped - this would be fairly easy to fix via wrapping on onNewPlayerShip
+-- If there is a use case people want this for I will add that happily
+
 -- TODO
 -- all these functions need error handling
 --addCommsReply
---SpaceObject:setCommsFunction
---SpaceObject:onDestroyed
 --Artifact:onCollision
 --Artifact:onPlayerCollision
 --Artifact:onPickUp
 --Artifact:onPickup
 --Mine:onDestruction
--- TODO - mirror the class hierachy (needed for the SpaceObject callbacks)
--- it would be nice if walking up the class hierarchy was moved out into a library
---SpaceObject
---    Artifact
---    Asteroid
---    BeamEffect
---    BlackHole
---    ElectricExplosionEffect
---    ExplosionEffect
---    Mine
---    MissileWeapon
---        EMPMissile
---        HVLI
---        HomingMissile
---        Nuke
---    Nebula
---    Planet
---    ScanProbe
---    ShipTemplateBasedObject
---    SupplyDrop
---    VisualAsteroid
---    WarpJammer
---    WormHole
---    Zone
--- note - its worth checking the gm create menu (or other ways) uses these functions rather than the C++ version
--- player ship is perticullary important
 -- it might be good to make some of these optional
 -- update and init might want a check before assuming they are present (and a warning if not?)
 -- check if objects have been created before wrapAllFunctions is called?
 -- prevent wrapAllFunctions being called more than once?
 -- the ShipTemplateBasedObject functions seem tempermental in applying - I need to investigate this
+-- check if no comms message set?

```

Comment:
I assume putting in code of this kind would change the question mark that comes back to something else? If you're going to put something like that in, just a lua line number would be nice so I can go find where either the communications message is not set or where the communications message gives an error